### PR TITLE
OpenVINO Execution Provider with 2023.2 support

### PR DESCRIPTION
**Pull request by sspintel**
- Add support for OpenVINO 2023.2
- num_of_threads provider option is mapped to the CPU device property inference_num_threads of the CPU plugin, so users can control the #threads used for inference by the CPU
- Logging in Debug mode now includes the runtime properties set for devices
- Fix issue in using external weights through OpenVINO

### CODE REVIEW

File: onnxruntime/core/session/provider_registration.cc [hunk 1]
```diff
@@ -84,11 +84,7 @@ ORT_API_STATUS_IMPL(OrtApis::SessionOptionsAppendExecutionProvider,
     status = create_not_supported_status();
 #endif
   } else if (strcmp(provider_name, "OpenVINO") == 0) {
-#if defined(USE_OPENVINO)
```
Comment (jywu-msft) [inline onnxruntime/core/session/provider_registration.cc:87]: why is this removed?
It can break builds that don't have openvino enabled. e.g. https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=1240589&view=logs&j=91d18614-091d-543c-f7ca-c2dad642e5d4&t=ce078c19-a5cd-536f-c923-6a49c8eb8ff8

Comment (sspintel) [reply onnxruntime/core/session/provider_registration.cc]: Hi George, This was a mistake and we've reverted these changes. However, we'll have to find an alternative soon to allow linking to signed OpenVINO dlls at runtime when the onnxruntime dlls weren't originally built with USE_OPENVINO.

**Review by jywu-msft — APPROVED** [review]
